identity/app/services/DiscusssionApiService.scala

```diff
@@ -0,0 +1,53 @@
+package services
+
+import play.api.libs.json.{Format, JsValue, Json}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import play.api.libs.ws.{WSClient, WSResponse}
+import utils.SafeLogging
+import cats.implicits._
+import cats.data.EitherT
+
+// A user will exist in Discussion if they have commented or recommended a comment.
+// DAPI does not expose recommendation stats by user, so we only rely on comments to decide whether or not a user in Discussion should have a public profile
+case class ProfileStats(status: String, comments: Int, pickedComments: Int) {
+  def hasComments: Boolean = comments > 0 // Will be zero for users who have recommended but not commented.
+}
+
+object ProfileStats {
+  implicit val format: Format[ProfileStats] = Json.format[ProfileStats]
+}
+
+case class DiscussionApiServiceException(message: String) extends Throwable
+
+class DiscussionApiService(wsClient: WSClient, config: conf.IdentityConfiguration)(implicit executionContext: ExecutionContext) extends SafeLogging {
+
+  private def GET(urlPath: String): Future[WSResponse] = {
+    wsClient
+      .url(config.discussionApiUrl + urlPath)
+      .withRequestTimeout(1.seconds)
```

Comment:
Is this the correct threshold for response from the API ? Could it be longer ?

Reply:
I've updated this to 3sec. Latency spikes in Discussion metrics have hit up to 1sec in past 2 weeks.